docs/design/thee_datahall.md

```diff
@@ -0,0 +1,197 @@
+The deployment pattern is similar to the [multi Kubernetes deployment](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/master/docs/manual/fault_domains.md#option-2-multi-kubernetes-replication) but we can't use the special key here since that would mean that the `locality_zone` field would be set to the Kubernetes clusters zone ID and for `three_data_hall` or `three_datacenter` we need at least 2 zones per `data_hall`/`data_center`.
+The initial deployment would be splitt into two phases:
+
+1. Deploy the seed cluster into one of the Kubernetes clusters and wait until it's fully reconciled.
+1. Copy the connection string and use it as `seedConnectionString` for the other two clusters and set `redundancy_mode: "three_data_hall"`.
+
+The initial cluster spec could look like this:
+
+```yaml
+apiVersion: apps.foundationdb.org/v1beta1
+kind: FoundationDBCluster
+metadata:
+  labels:
+    cluster-group: test-cluster
+  name: test-cluster
+spec:
+  version: 6.2.30
+  instanceIDPrefix: az1
+  databaseConfiguration:
+    redundancy_mode: "triple"
+  localities:
+  - key: "locality_data_hall"
+    valueFromNode: "topology.kubernetes.io/zone"
+```
+
+With this configuration we will use the default fault domain the Kubernetes nodes as `locality_zone`.
+When the clutser is reconciled we can create the `FoundationDBCluster` spec in the other two Kubernetes clusters and set the `redundancy_mode: "three_data_hall"`:
+
+```yaml
+apiVersion: apps.foundationdb.org/v1beta1
+kind: FoundationDBCluster
+metadata:
+  labels:
+    cluster-group: test-cluster
+  name: test-cluster
+spec:
+  version: 6.2.30
+  instanceIDPrefix: $az
+  seedConnectionString: $connectionString
+  databaseConfiguration:
+    redundancy_mode: "three_data_hall"
+  localities:
+  - key: "locality_data_hall"
+    valueFromNode: "topology.kubernetes.io/zone"
+```
+
+The processes in the other Kubernetes clusters will join the current `FoundationDBCluster`.
+Once enough processes joined the cluster one of the operator will select 5 Coordinators span across 3 `data_halls`.
+
+### Coordinator selection
+
+The coordinator selecion must be adjusted to select at most 2 coordinators in the same `data_hall` or `datacenter` depending on the redundancy mode.
+Currently the selection happens in the [cluster_controller](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/master/controllers/cluster_controller.go#L1283-L1292) with a small extension we can support the coordinator selection:
+
+```go
+func getHardLimits(cluster *fdbtypes.FoundationDBCluster) map[string]int {
+    // Multi region cluster (HA)
+    if cluster.Spec.UsableRegions > 1 {
+        return map[string]int{
+            fdbtypes.FDBLocalityZoneIDKey: 1, 
+            fdbtypes.FDBLocalityDCIDKey: int(math.Floor(float64(cluster.DesiredCoordinatorCount()) / 2.0))
+        }
+    }
+
+    // Single region cluster (non-HA)
+    req := map[string]int{fdbtypes.FDBLocalityZoneIDKey: 1}
```

Comment:
```suggestion
    req := map[string]int{fdbtypes.FDBLocalityZoneIDKey: 1}
    if cluster.Spec.UsableRegions > 1 {
        req[fdbtypes.FDBLocalityDCIDKey] = int(math.Floor(float64(cluster.DesiredCoordinatorCount()) / 2.0))
    }
```

This change allows to use `three_data_hall` with multi region clusters. We only need an additional check if we are using multi region and throw an error if `three_datacenter` is set.